pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveMetadataFetcher.java

```diff
@@ -62,11 +74,24 @@ public HiveMetadataFetcher(InputData md) {
 
         List<Metadata> metadataList = new ArrayList<Metadata>();
 
+        if(tblsDesc.size() > 1) {
```

Comment:
What if there are 2 tables but both are not supported? in that case the function will return 0 tables. Is that an acceptable scenario? if so, perhaps add a test to cover it?

Did you consider changing the condition to have at least one 'good' table - check in the end of the loop if there metadataList is not empty. If it is, throw an exception.